### PR TITLE
Fix ESM & Commonjs exports definitions

### LINKED ISSUE #105: build es6 version of the sql-formatter
In order to use sql-formatter in an es6 project I had to import it in this format `import sqlFormatter from 'sql-formatter/src/sqlFormatter.js';` and then download all the dependencies for sql-formatter which is currently only "lodash" in its es6 version.

I would like to use an es6 version of the sql-formatter at runtime without any bundling/packaging step so that we can import it in a proper es6 format import `import sqlFormatter from "sql-formatter";`

### PR DESCRIPTION
**Pull request by nene**
Fixes #105

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,11 +3,15 @@
...
-  "main": "dist/sql-formatter.min.js",
+  "main": "dist/sql-formatter.min.cjs",
   "module": "lib/index.js",
   "types": "lib/src/index.d.ts",
   "exports": {
-    ".": "./lib/index.js"
+    ".": {
+      "import": "./lib/index.js",
+      "require": "./dist/sql-formatter.min.cjs",
+      "types": "./lib/src/index.d.ts"
```
Comment (JounQin) [inline package.json:13]: > the "types" condition should always come first in "exports"

See https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/#package-json-exports-imports-and-self-referencing

**Review by JounQin — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (nene) [reply package.json:13]: Thanks a lot. Wouldn't have guessed this gotcha on my own.

File: package.json [hunk 2]
```diff
@@ -3,11 +3,15 @@
...
+    }
   },
   "bin": {
     "sql-formatter": "./bin/sql-formatter-cli.js"
```
Comment (JounQin) [inline package.json:17]: The bin file has not been migrated to ESM, so `.cjs` suffix is still required, and

```diff
- const { format, supportedDialects } = require('../lib/index');
+ const { format, supportedDialects } = require('./dist/sql-formatter.min.cjs');
```

**Review by JounQin — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (nene) [reply package.json]: Thanks again.

I'm really not good at this ESM conversion :(

**Review by JounQin — APPROVED** [review]

File: webpack.common.js [hunk 3]
```diff
@@ -4,7 +4,7 @@ export default {
   entry: './src/index.ts',
   output: {
     path: path.resolve('./dist'),
-    filename: 'sql-formatter.js',
+    filename: 'sql-formatter.cjs',
     library: { type: 'module' },
```
[the remark this answers is not in the document]
Comment (nene) [reply webpack.common.js:8]: Indeed, that's clearly incorrect. Not sure why the webpack build was changed to module.

**Pull request merged**